components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -826,7 +832,10 @@ private boolean isAskPasswordFlow(String recoveryScenario, User user) throws Ide
             isPendingAskPasswordState = IdentityRecoveryConstants.PENDING_ASK_PASSWORD.equals(accountState);
         }
 
-        return RecoveryScenarios.ASK_PASSWORD.equals(RecoveryScenarios.getRecoveryScenario(recoveryScenario)) &&
+        RecoveryScenarios recoveryScenarioType = RecoveryScenarios.getRecoveryScenario(recoveryScenario);
```

Comment:
recoveryScenario

Reply:
There is an argument passed as `recoveryScenario` hence the local variable is defined as `recoveryScenarioType`